presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java

```diff
@@ -361,6 +364,7 @@ public void configure(Binder moduleBinder)
                     {
                         configBinder(moduleBinder).bindConfig(ResourceManagerConfig.class);
                         moduleBinder.bind(ClusterStatusSender.class).to(ResourceManagerClusterStatusSender.class).in(Scopes.SINGLETON);
+                        moduleBinder.bind(MemoryManagerService.class).to(ClusterMemoryManagerService.class).in(Scopes.SINGLETON);
```

Comment:
should this binding be based on flag if RM enabled then only enable it?

Reply:
That's what it's doing (see line 359).

Reply:
This injection causing issue as even worker will have RM enabled flag and that leads to injecting this part to worker as well. We need to inject it only for Coordinator.